pkg/controller/windowsmachineconfig/nodeconfig/nodeconfig.go

```diff
@@ -48,6 +51,53 @@ func (nc *nodeConfig) Configure() error {
 	if err := nc.handleCSRs(); err != nil {
 		return errors.Wrap(err, "handling CSR for the given node failed")
 	}
+
+	// Now that basic kubelet configuration is complete, configure networking in the node
+	if err := nc.configureNetwork(); err != nil {
+		return errors.Wrap(err, "configuring node network failed")
+	}
+	log.Info("VM has been configured as a worker node", "VM ID", nc.GetCredentials().GetInstanceId())
+	return nil
+}
+
+// configureNetwork configures k8s networking in the node
+func (nc *nodeConfig) configureNetwork() error {
+	if nc.GetCredentials() == nil {
+		return fmt.Errorf("nil credentials for VM")
+	}
+
+	if nc.GetCredentials().GetIPAddress() == "" {
+		return fmt.Errorf("empty IP for VM: %v", nc.GetCredentials())
+	}
+
+	node, err := nc.getNode(nc.GetCredentials().GetIPAddress())
+	if err != nil {
+		return errors.Wrap(err, fmt.Sprintf("error getting node object for VM %s",
+			nc.GetCredentials().GetInstanceId()))
+	}
+
+	// Wait until the node object has the hybrid overlay subnet annotation. Otherwise the hybrid-overlay will fail to
+	// start
+	if err = nc.waitForNodeAnnotation(node.GetName(), HybridOverlaySubnet); err != nil {
+		return errors.Wrap(err, fmt.Sprintf("error waiting for %s node annotation for %s", HybridOverlaySubnet,
+			node.GetName()))
+	}
+
+	// NOTE: Investigate if we need to introduce a interface wrt to the VM's networking configuration. This will
+	// become more clear with the outcome of https://issues.redhat.com/browse/WINC-343
+
+	// Configure the hybrid overlay in the Windows VM
```

Comment:
nit: Writing this down here so that we don't forget. We can have another networkConfig struct which in composed within nodeConfig struct
```suggestion
	type networkConfig struct {
          kubeclient
          WindowsVM
          neededAnnotations []string
	}
```

With this when we move to interface definition it will be much more cleaner as to which methods are needed. No need to address this now.